src/rust/ensogl/lib/text/src/buffer/data/unit.rs

```diff
@@ -24,29 +25,74 @@ pub use traits::*;
 // === Bytes ===
 // =============
 
-types::unit! {
+unit! {
 /// An offset in the buffer in bytes.
-Bytes::bytes(usize)
+Bytes::bytes(i32)
+}
+
+impl Bytes {
+    /// Saturating conversion to `usize`.
+    pub fn as_usize(self) -> usize {
+        self.value.max(0) as usize
+    }
 }
 
 impl<T:Into<Bytes>> bytes::Into for Range<T> {
     type Output = Range<Bytes>;
     fn bytes(self) -> Self::Output {
-        let start = self.start.bytes();
-        let end   = self.end.bytes();
+        let start = self.start.into();
+        let end   = self.end.into();
         Range {start,end}
     }
 }
 
+impl From<usize> for Bytes {
+    fn from(t:usize) -> Self {
+        (t as i32).into()
+    }
+}
+
+impl From<&usize> for Bytes {
+    fn from(t:&usize) -> Self {
+        (*t as i32).into()
+    }
+}
+
 
 
 // ============
 // === Line ===
 // ============
 
-types::unit! {
+// TODO: Improvement idea. Create `i32Saturated` type which will have all operations saturated.
+//       This will make this unit safer.
```

Comment:
To what safety issues it can lead when working with columns or liens in text? Could you imagine a single situation where panicking there would be better than saturation? Moreover, we are using `ssaturating_add`, `saturating_sub` etc functions just everywhere in our code now. Making the described improvement just simplifies the code and makes sure no one forgets about using saturating version of the operators.

Reply:
God, nvm, I confised "saturating" with "overflowing" :facepalm: